src/service/manager.js

```diff
@@ -102,10 +104,22 @@ const Manager = GObject.registerClass({
 
     get certificate() {
         if (this._certificate === undefined) {
-            this._certificate = Gio.TlsCertificate.new_for_paths(
-                GLib.build_filenamev([Config.CONFIGDIR, 'certificate.pem']),
-                GLib.build_filenamev([Config.CONFIGDIR, 'private.pem']),
-                null);
+            const app = Gio.Application.get_default();
+            try {
+                this._certificate = Gio.TlsCertificate.new_for_paths(
+                    GLib.build_filenamev([Config.CONFIGDIR, 'certificate.pem']),
+                    GLib.build_filenamev([Config.CONFIGDIR, 'private.pem']),
+                    null);
+                if (this.settings.get_boolean('missing-openssl'))
+                    app?.withdraw_notification('gsconnect-missing-openssl');
+                this.settings.set_boolean('missing-openssl', false);
```

Comment:
Suppose I could've saved a couple of CPU cycles if I'd put this settings  change _inside_ the `if()`. Oh well.